src/session.ts

```diff
@@ -497,7 +502,18 @@ export class Session extends common.GrpcServiceObject {
     queryOptions?: google.spanner.v1.ExecuteSqlRequest.IQueryOptions,
     requestOptions?: Pick<IRequestOptions, 'transactionTag'>
   ) {
-    return new Transaction(this, undefined, queryOptions, requestOptions);
+    const span = getActiveOrNoopSpan();
+    span.addEvent('Creating Transaction');
+    const transaction = new Transaction(
+      this,
+      undefined,
+      queryOptions,
+      requestOptions
+    );
+    span.addEvent('Transaction Creation Done', {
```

Comment:
Same here, remove all the create transaction and Transaction Creation Done event and keep at only one place, recommended in above comment